Add files via upload

aquiladayc opened the pull request:
#### What this PR does / why we need it:
- Clean up example files of values.yaml
- Having minimal setups for each use cases

#### Special notes for your reviewer:
Hi @JacksonDavenport, can you review please? 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [ ] Chart Version bumped
- [ ] `CHANGELOG.md` has been updated
- [ ] Variables are documented in the `README.md`

estherk15 commented on examples/datadog/README.md:
```diff
@@ -0,0 +1,3 @@
+## Examples of values.yaml for Datadog Helm Chart 
+Examples of values.yaml with minimal configurations for each use cases.
+For the full list of the Datadog Helm Chart’s configurable parameters and their default values, see the [Datadog Helm repository README](https://github.com/DataDog/helm-charts/tree/main/charts/datadog) and [values.yaml](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml) as a reference. 
```
```suggestion
For the full list of the configurable parameters and default values, see the [Datadog Helm repository README](https://github.com/DataDog/helm-charts/tree/main/charts/datadog) and [values.yaml](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml). 
```

estherk15 reviewed (approved):
Minor copy suggestion, looks good!

JacksonDavenport commented on examples/datadog/agent_otlp.yaml:
```diff
@@ -0,0 +1,24 @@
+    receiver:
+      protocols:
+        grpc:
+          enabled: false
+          endpoint: "0.0.0.0:4317"
+          useHostPort: true
+        http:
+          enabled: false
+          endpoint: "0.0.0.0:4318"
+          useHostPort: true
```
For otlp these (`...grpc.enabled` and `...http.enabled`) can be set to `true`. To ensure the Agent is listening for them.

JacksonDavenport commented on examples/datadog/agent_otlp.yaml:
```diff
@@ -0,0 +1,24 @@
+  appKey: <DATADOG_APP_KEY>
+  clusterName: <CLUSTER_NAME>
+  tags: []
+  apm:
+    # Enable APM over Socket (Unix Socket or windows named pipe)
+    socketEnabled: true
+    socketPath: /var/run/datadog/apm.socket
+    hostSocketPath: /var/run/datadog/
```
Nit: APM isn't technically necessary for OTLP. As that data will actually go through the `agent` instead of the `trace-agent`. I guess generally there likely are a lot of customers using both though so this isn't going to harm anything but isn't needed.

aquiladayc replied:
To keep it minimal, I would remove it.

JacksonDavenport commented on examples/datadog/agent_on_gke_autopilot_values.yaml:
```diff
@@ -0,0 +1,53 @@
+    enabled: true
+  systemProbe:
+    enableTCPQueueLength: false
+    enableOOMKill: true
```
`system-probe` isn't supported with GKE Autopilot, can set `datadog.systemProbe.enableOOMKill: false` to avoid deploying that.

aquiladayc replied:
Removed it to not deploy it. Tested in my env.

JacksonDavenport commented on examples/datadog/agent_on_aks_values.yaml:
```diff
@@ -19,19 +16,18 @@ datadog:
-    collectDNSStats: false
+    enableOOMKill: false
+    collectDNSStats: true
 providers:
     aks:
       enabled: true
```
```suggestion
providers:
  aks:
    enabled: true
```
Fix whitespace

JacksonDavenport commented on examples/datadog/agent_on_aks_values_windows.yaml:
```diff
@@ -1,24 +1,22 @@
   processAgent:
     enabled: true
-    processCollection: false
+providers:
+    aks:
+      enabled: true
```
```suggestion
providers:
  aks:
    enabled: true
```
Fix whitespace

JacksonDavenport commented on examples/datadog/agent_confd.yaml:
```diff
@@ -0,0 +1,25 @@
+  appKey: <DATADOG_APP_KEY>
+  clusterName: <CLUSTER_NAME>
+  tags: []
+  # Each key becomes a file in /conf.d and overwrite default settings
```
It may be worthwhile to include a snippet for `datadog.ignoreAutoConf`, as while these samples are for the `kubelet` and `kube_proxy` integration if a user creates a `redisdb.yaml` technically this doesn't overwrite the `redisdb.d/auto_conf.yaml`. 

Even leaving blank but exposing that might be good here:
```yaml
  ## ref: https://docs.datadoghq.com/agent/faq/auto_conf/
  ignoreAutoConfig: []
  #  - redisdb
  #  - kubernetes_state
```

aquiladayc replied:
Makes sense. Include the option but remove `- redisdb` and `- kubernetes_state` to keep it simple. Instead, added a comment to explain this option (copied from public doc).
And document link is updated to point more specific section.

JacksonDavenport reviewed (changes requested):
Some great changes in here to add some more examples. Added some comments and a few suggestions.

aquiladayc commented:
Hi @JacksonDavenport, reflected you feedback. Can you have a look when you have a chance?
Btw, this branch is out-of-date but I'm leaving it for now. We can include those change before we merge this branch to main.

CharlyF commented on examples/datadog/agent_apm.yaml:
```diff
@@ -0,0 +1,33 @@
+    ## Optional configurations for trace-agent container
+    traceAgent:
+      env:
+      envFrom: []
+      logLevel: 
+      resources: {}
```
let's only give examples with populated fields?

aquiladayc replied:
👍

CharlyF commented on examples/datadog/agent_ksm_core.yaml:
```diff
@@ -0,0 +1,29 @@
+## Enable Cluster check runner, if dispatching KSM Core Check
+## See: https://docs.datadoghq.com/containers/cluster_agent/clusterchecksrunner/?tab=helm#pagetitle
+clusterChecksRunner:
+  enabled: false
```
can we add the config (commented) to run KSM core as a distributed cluster check with the cluster check runner/DCA?

aquiladayc replied:
Sure, I added a new config. And removed comments and cluster check runner section from this.

CharlyF commented on examples/datadog/agent_on_openshift_values.yaml:
```diff
@@ -6,25 +6,15 @@
   # If you have proper CA, you can switch it to true
   kubelet:
     tlsVerify: false
-  confd:
```
just confirming that we are removing it because the check is now autodetected?

aquiladayc replied:
Yes we wanted to keep it simple by removing unnecessary fields. As I tested in my sandbox, it seems working properly without it. 
But do you think it better to keep it?

CharlyF reviewed (changes requested):
very helpful contribution - Left a few questions and a general recommendation of not adding fields that are empty (unless there is a reason for it).

CharlyF reviewed (approved):
lgtm, just needs a rebase. Thank you!

aquiladayc commented:
Hey @CharlyF, I rebased and updated my branch. Can you review? It's the first time I did it (T2 team is not really familiar with this), please let me know if anything should be fixed.
